src/store/levels/reducer.js

```diff
@@ -0,0 +1,40 @@
+import { START_LEVEL, ROTATE_TILE } from "./action";
+
+const initialState = { currentLevel: 0, tiles: null };
+
+export default (state = initialState, action = {}) => {
+  switch (action.type) {
+    case START_LEVEL: {
+      return {
+        ...state,
+        currentLevel: +1,
+        tiles: action.payload
+      };
+    }
+
+    case ROTATE_TILE: {
+      const { x, y, rotatedTile } = action.payload;
+      const newState = {};
+      newState.tiles = state.tiles.map((currentTile, index) => {
```

Comment:
in this case it might be a bit dumb, i know. but as a general rule try to not mutate. so:
```
const tiles = state.tiles.map(.....)
const newState = { tiles }; 
// or just return
return { tiles }
```